File: CppClient/Game/ai.cpp
#include "ai.h"

using namespace koala::chillin::client;
using namespace ks::models;


AI::AI(World *world): TurnbasedAI<World*>(world)
{
}

void AI::decide()
{
    for (int i = 0; i < 3; i++)
        for (int j = 0; j < 3; j++)
            if (world->board()[i][j] == ECell::Empty)
            {
                ks::commands::Place cmd;
                cmd.x(j);
                cmd.y(i);
                sendCommand(&cmd);
                return ;
            }
}

File: JavaClient/src/ai/AI.java
package ai;

import team.koala.chillin.client.TurnbasedAI;
import ks.KSObject;
import ks.models.*;
import ks.commands.*;


public class AI extends TurnbasedAI<World, KSObject> {
	
	public AI(World world) {
		super(world);
	}

	@Override
	public void decide() {
		for(int i = 0; i < world.getBoard().length; i++)
			for(int j = 0; j < world.getBoard()[0].length; j++)
				if (world.getBoard()[i][j] == ECell.Empty) {
					final int placeX = j;
					final int placeY = i;
					this.sendCommand(new Place() {{ x = placeX; y = placeY; }});
					return;
				}
	}
}
